.github/workflows/CI.yml

```
@@ -47,7 +47,11 @@ jobs:
       run: cargo fmt --all -- --check
 
     - name: Build Project
-      run: cargo build --workspace --all-features
+      run: cargo build --workspace --all-features -D warning
 
     - name: Run Tests
       run: cargo nextest run --workspace --all-features
+    
+    -name: check for warnings in code 
```

```suggestion
    - name: Lints checks
```
Don't forget to add space between the dash and **name**.

---

Yes it is a good idea